src/store/features/contacts/contactsSlice.ts

```diff
@@ -0,0 +1,85 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import axios from 'axios';
+import { API_URL } from "../../../env";
+import { Contact } from "../../../types";
+
+export const getContacts = createAsyncThunk('contacts/fetchContacts', async (args: { country: string, query: string, page: number, onlyEven: number }, { rejectWithValue }) => {
+        const { country, query, page, onlyEven } = args;
+
+        try {
+            const result = await axios.get<GetContactsResult>(`${API_URL}/contacts?country=${country}&query=${query}&onlyEven=${onlyEven}&page=${page}`)
+
+            return {
+                data: result.data
+            }
+        } catch (err: any) {
+            return rejectWithValue(err.message)
+        }
+    }
+)
+
+
+interface GetContactsResult {
+    contacts: Contact[],
+    count: number,
+    page: number
+    next: boolean,
+}
+
+interface Contacts {
```

Comment:
Sa pui si State aici la final pentru a nu se confunda cu un eventual tip de data folosit in aplicatie